homeassistant/components/modbus/__init__.py

```diff
@@ -183,19 +183,23 @@ def number(value: Any) -> int | float:
         ),
         vol.Optional(CONF_COMMAND_OFF, default=0x00): cv.positive_int,
         vol.Optional(CONF_COMMAND_ON, default=0x01): cv.positive_int,
-        vol.Optional(CONF_VERIFY): {
-            vol.Optional(CONF_ADDRESS): cv.positive_int,
-            vol.Optional(CONF_INPUT_TYPE): vol.In(
-                [
-                    CALL_TYPE_REGISTER_HOLDING,
-                    CALL_TYPE_DISCRETE,
-                    CALL_TYPE_REGISTER_INPUT,
-                    CALL_TYPE_COIL,
-                ]
-            ),
-            vol.Optional(CONF_STATE_OFF): cv.positive_int,
-            vol.Optional(CONF_STATE_ON): cv.positive_int,
-        },
+        vol.Optional(CONF_VERIFY): vol.Any(
+            {
+                vol.Optional(CONF_ADDRESS): cv.positive_int,
+                vol.Optional(CONF_INPUT_TYPE): vol.In(
+                    [
+                        CALL_TYPE_REGISTER_HOLDING,
+                        CALL_TYPE_DISCRETE,
+                        CALL_TYPE_REGISTER_INPUT,
+                        CALL_TYPE_COIL,
+                    ]
+                ),
+                vol.Optional(CONF_STATE_OFF): cv.positive_int,
+                vol.Optional(CONF_STATE_ON): cv.positive_int,
+                vol.Optional("allow_empty_verify", default=0): cv.positive_int,
+            },
+            None,
```

Comment:
Why do we need to allow `None`?

Reply:
Because all attributes are optional (in most cases the values are identical to the write parameters), meaning the user will write
ˋˋˋ
switch:
       ....
       verify:
ˋˋˋ
remark “verify: “ is empty == None.
to indicate he wants the switch verified, with same address etc.
This is one item I think will be a lot easier when we move to config_flow.

Reply:
We should probably explain this a bit more in the docs. There's just two different kinds of examples and no explanation that both forms are possible, that I can see.